beacon-chain/rpc/apimiddleware/custom_handlers.go

```diff
@@ -0,0 +1,159 @@
+package apimiddleware
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/prysmaticlabs/prysm/shared/gateway"
+	"github.com/prysmaticlabs/prysm/shared/grpcutils"
+)
+
+type sszConfig struct {
+	sszPath      string
+	fileName     string
+	responseJson sszResponseJson
+}
+
+func handleGetBeaconStateSsz(m *gateway.ApiProxyMiddleware, endpoint gateway.Endpoint, writer http.ResponseWriter, request *http.Request) (handled bool) {
+	config := sszConfig{
+		sszPath:      "/eth/v1/debug/beacon/states/{state_id}/ssz",
+		fileName:     "beacon_state.ssz",
```

Comment:
Is it spec to call the file name like this? It might be useful to include the slot number in the filename, perhaps.

Reply:
There is no name enforced by the spec. Although I agree that `beacon_state.ssz` is very generic, this function does not have access to any information that could help here. If it's OK, I would like to leave it as it is.